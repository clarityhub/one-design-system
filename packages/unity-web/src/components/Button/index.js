export { default, EmotionButton } from './Button';