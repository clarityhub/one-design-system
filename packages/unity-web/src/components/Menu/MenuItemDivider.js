import styled from '@emotion/styled';
import colors from '@clarityhub/unity-core/lib/colors';

const MenuItemDivider = styled.hr`
    width: 100%;
    height: 0;
    border-top: 1px solid ${colors.gray.default};
`;

export default MenuItemDivider;
