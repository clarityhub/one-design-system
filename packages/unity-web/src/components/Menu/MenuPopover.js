/** @jsx jsx */

import React, { Component } from 'react';
import { jsx, css } from '@emotion/core';
import styled from '@emotion/styled';
import { arrayOf, node } from 'prop-types';
import { variants } from '@clarityhub/unity-core/lib/typography';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';

import { offset, noOverflow } from '../../utilities/position';

const menuPopoverWrapper = () => css`
    background-color: ${colors.white.default};
    border-radius: ${borders.borderRadius.rem}rem;
    box-shadow: ${colors.shadow.default};
    position: absolute;
    z-index: 99999999;
    margin-top: 1rem;
    min-width: 13rem;
    padding: 0.4rem 0;

    ${variants.button.string}

    font-size: 1rem;

    ul {
        list-style: none;
        margin: 0;
        padding: 0;
    }

    li {
        margin: 0;
        padding: 0;
    }
`;

const Arrow = styled.div`
    position: absolute;
    width: 0; 
    height: 0; 
    border-left: 0.5rem solid transparent;
    border-right: 0.5rem solid transparent;
    border-bottom: 0.5rem solid ${colors.white.default};
    top: -0.5rem;
    transform: translateX(-50%);
`;

class MenuPopover extends Component {
    static propTypes = {
    	items: arrayOf(node).isRequired,
    	target: node,
    }

    state = {
    	position: {},
    	arrow: {},
    }

    componentDidMount() {
    	const position = this.getPosition(this.props, this.popover);
    	this.setState({
    		position,
    		arrow: this.getArrowPosition(this.props, position),
    	});
    }

    componentWillUpdate(nextProps) {
    	if (nextProps.left !== this.props.left ||
            nextProps.target !== this.props.target) {
            
    		const position = this.getPosition(nextProps, this.popover);
    		this.setState({
    			position,
    			arrow: this.getArrowPosition(nextProps, position),
    		});
    	}
    }

    getPosition = (props, el) => {
    	let pos = { ...props.position };
    	if (el) {
    		// Try to center
    		pos.left = pos.left - (offset(el).width / 2) + (offset(props.target).width / 2);
    	}

    	// Push over if it would go off the page
    	return noOverflow(pos, el);
    }

    getArrowPosition = (props, menuPosition) => {
    	if (props.target) {
    		const targetPos = offset(props.target);

    		const left = targetPos.left - menuPosition.left + (targetPos.width / 2);

    		return {
    			top: 0,
    			left,
    		};
    	} else {
    		return {
    			top: 0,
    			left: 0,
    		};
    	}
    }

    render() {
    	const { position, arrow } = this.state;
    	const { items, innerRef } = this.props;

    	return (
    		<div
    			ref={(ref) => { this.popover = ref; innerRef(ref);}}
    			css={menuPopoverWrapper}
    			style={{
    				top: `${position.bottom || 0}px`,
    				left: `${position.left || 0}px`,
    			}}
    		>
    			<Arrow
    				style={{
    					left: `${arrow.left || 0}px`,
    				}}
    			/>
    			<ul>
    				{items.map((item, i) => (
    					<li key={i}>
    						{item}
    					</li>
    				))}
    			</ul>
    		</div>
    	);
    }
}

export default React.forwardRef((props, ref) => <MenuPopover innerRef={ref} {...props} />);
