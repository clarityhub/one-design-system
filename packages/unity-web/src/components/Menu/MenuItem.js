/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import colors from '@clarityhub/unity-core/lib/colors';

import Button from '../Buttons';

const MenuItem = props => (
	<Button
		text
		css={css`
            border-radius: 0;
            text-align: left;
            text-transform: none;
            min-width: 150px;
            width: 100%;

            &:hover {
                background-color: ${colors.muted.default};
            }
        `}
		{...props}
	/>
);

export default MenuItem;
