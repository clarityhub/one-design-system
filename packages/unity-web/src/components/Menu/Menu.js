/** @jsx jsx */

import { Component } from 'react';
import { func, node, arrayOf } from 'prop-types';
import styled from '@emotion/styled';
import { jsx } from '@emotion/core';

import { offset } from '../../utilities/position';
import MenuPopover from './MenuPopover';
import MenuPortal from './MenuPortal';

const MenuWrapper = styled.div`
    position: relative;
`;

const MenuTarget = styled.div`
    position: relative;
`;

/**
 * When you click on the {children} wrapper,
 * a menu will appear below the {children}'s position.
 * 
 * The menu's position will remain within the window.
 * 
 * The menu will close when you click outside of the menu
 */
export default class Menu extends Component {
    static propTypes = {
    	children: func.isRequired,
    	items: arrayOf(node).isRequired,
    }

    state = {
    	open: false,
    }

    componentDidMount() {
    	this._onClickOutside = this.onClickOutside.bind(this);
    	this._onKeyUp = this.onKeyUp.bind(this);

    	document.addEventListener('click', this._onClickOutside);
    	document.addEventListener('keyup', this._onKeyUp);
    }

    componentWillUnmount() {
    	document.removeEventListener('click', this._onClickOutside);
    	document.removeEventListener('keyup', this._onKeyUp);
    }

    onClickOutside = (e) => {
    	if (this.menu && this.doneOpening) {        
    		if (this.state.open) {
    			const target = e.target;
    			// check if target is within (or is) the menu
    			const clickOutsideMenu = this.menu !== target && !this.menu.contains(target);
    			if (clickOutsideMenu) {
    				this.close();
    			}
    		}
    	}
    }

    onKeyUp = (e) => {
    	if (this.state.open) {
    		if (e.key === 'Escape') {
    			this.close();
    		}
    	}
    }

    open = () => {
    	this.doneOpening = false;

    	let position = {
    		top: 0,
    		left: 0,
    	};

    	if (this.target) {
    		position = offset(this.target);
    	}

    	this.setState(({ open }) => ({
    		open: !open,
    		position,
    	}), () => {
    		setTimeout(() => {
    			this.doneOpening = true;
    		}, 100);
    	});
    }

    close = () => {
    	this.setState({
    		open: false,
    	});
    }

    render() {
    	const { children, items } = this.props;
    	const { open, position } = this.state;

    	return (
    		<MenuWrapper>
    			{open && (
    				<MenuPortal>
    					<MenuPopover
    						ref={ref => this.menu = ref}
    						items={items}
    						position={position}
    						target={this.target}
    					/>
    				</MenuPortal>
    			)}

    			<MenuTarget ref={ref => this.target = ref}>
    				{children({
    					open: this.open,
    				})}
    			</MenuTarget>
    		</MenuWrapper>
    	);
    }
}
