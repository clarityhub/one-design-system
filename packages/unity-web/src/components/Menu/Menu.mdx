---
name: Menu
menu: Components
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import { Menu, MenuItem, MenuItemDivider } from './';
import Button from '../Buttons';

# Menu

Show a pop-up menu in navigation or when there are many options
a user can select.

Menus will always appear fixed to the bottom-left of the child
element, but will fix to the bottom-right if the window width
would be less than the width of the popover.

## \<Menu /\> component

```js
import { Menu, MenuItem, MenuItemDivider } from '@clarityhub/unity-web/lib/components/Menu';
```

## Prop Types

### Menu Props

<PropsTable of={Menu} />

### MenuItem Props

All `Button` props can also be passed to the `MenuItem`.

<PropsTable of={MenuItem} />

### MenuItemDivider Props

<PropsTable of={MenuItemDivider} />

## Examples

<Playground>
    <Menu
        items={[
            <MenuItem>Item 1</MenuItem>,
            <MenuItem>Item 2</MenuItem>,
            <MenuItem>Item 3</MenuItem>,
            <MenuItemDivider />,
            <MenuItem>Long Item 4</MenuItem>,
        ]}
    >
        {({ open }) => (
            <Button onClick={open}>Open Menu</Button>
        )}
    </Menu>
</Playground>

## Custom Content

<Playground>
    <Menu
        content={
            <div>
                <Button>Test</Button>
            </div>
        }
    >
        {({ open }) => (
            <Button onClick={open}>Open Menu</Button>
        )}
    </Menu>
</Playground>
