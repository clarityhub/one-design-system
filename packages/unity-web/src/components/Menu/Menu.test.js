import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Button from '../Button';
import Menu from './Menu';
import MenuItem from './MenuItem';
import MenuItemDivider from './MenuItemDivider';

describe('<Menu />', () => {
	const instance = {
		getBoundingClientRect() {
			return {
				top: 0,
				left: 0,
				right: 0,
				bottom: 0,
				height: 0,
				outerHeght: 0,
				width: 0,
			};
		},
	};
	function createNodeMock() {
		return instance;
	}

	it('renders a button', () => {
		const node = (
			<Menu
				inline
				noPortal
				items={[
					<MenuItem>Item 1</MenuItem>,
					<MenuItem>Item 2</MenuItem>,
					<MenuItem>Item 3</MenuItem>,
					<MenuItemDivider />,
					<MenuItem>Long Item 4</MenuItem>,
				]}
			>
				{({ open }) => (
					<Button onClick={open}>Open Menu</Button>
				)}
			</Menu>
		);
		const tree = renderer.create(node, { createNodeMock }).toJSON();

		expect(tree).toMatchSnapshot();
	});
    
	it('renders menu items', () => {
		const node = (
			<Menu
				noPortal
				inline
				items={[
					<MenuItem>Item 1</MenuItem>,
					<MenuItem>Item 2</MenuItem>,
					<MenuItem>Item 3</MenuItem>,
					<MenuItemDivider />,
					<MenuItem>Long Item 4</MenuItem>,
				]}
			>
				{({ open }) => (
					<Button onClick={open}>Open Menu</Button>
				)}
			</Menu>
		);
		const component = renderer.create(node, { createNodeMock });

		act(component.root.findByType('button').props.onClick);

		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});


