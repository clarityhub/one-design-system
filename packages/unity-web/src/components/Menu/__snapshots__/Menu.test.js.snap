// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders a button 1`] = `
.emotion-4 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-0 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus,
.emotion-0.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-0:disabled,
.emotion-0.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-0 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-0.-default,
.emotion-0.-primary,
.emotion-0.-danger,
.emotion-0.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-0.-default.-outline,
.emotion-0.-primary.-outline,
.emotion-0.-danger.-outline,
.emotion-0.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-0.-default.-text,
.emotion-0.-primary.-text,
.emotion-0.-danger.-text,
.emotion-0.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-0.-default:hover,
.emotion-0.-primary:hover,
.emotion-0.-danger:hover,
.emotion-0.-white:hover,
.emotion-0.-default.active,
.emotion-0.-primary.active,
.emotion-0.-danger.active,
.emotion-0.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-0.-default:hover.-outline,
.emotion-0.-primary:hover.-outline,
.emotion-0.-danger:hover.-outline,
.emotion-0.-white:hover.-outline,
.emotion-0.-default.active.-outline,
.emotion-0.-primary.active.-outline,
.emotion-0.-danger.active.-outline,
.emotion-0.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-0.-default:hover.-text,
.emotion-0.-primary:hover.-text,
.emotion-0.-danger:hover.-text,
.emotion-0.-white:hover.-text,
.emotion-0.-default.active.-text,
.emotion-0.-primary.active.-text,
.emotion-0.-danger.active.-text,
.emotion-0.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-0.-default:disabled,
.emotion-0.-primary:disabled,
.emotion-0.-danger:disabled,
.emotion-0.-white:disabled,
.emotion-0.-default.disabled,
.emotion-0.-primary.disabled,
.emotion-0.-danger.disabled,
.emotion-0.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-0.-default:disabled.-outline,
.emotion-0.-primary:disabled.-outline,
.emotion-0.-danger:disabled.-outline,
.emotion-0.-white:disabled.-outline,
.emotion-0.-default.disabled.-outline,
.emotion-0.-primary.disabled.-outline,
.emotion-0.-danger.disabled.-outline,
.emotion-0.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-0.-default:disabled.-text,
.emotion-0.-primary:disabled.-text,
.emotion-0.-danger:disabled.-text,
.emotion-0.-white:disabled.-text,
.emotion-0.-default.disabled.-text,
.emotion-0.-primary.disabled.-text,
.emotion-0.-danger.disabled.-text,
.emotion-0.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-0.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-0.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-0.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-0.-text:hover,
.emotion-0.-text.active {
  border-color: transparent;
}

.emotion-0.-text.disabled,
.emotion-0.-text:disabled {
  border: 1px solid transparent;
}

.emotion-0.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-0.disabled,
.emotion-0:disabled {
  cursor: not-allowed;
}

<div
  className="emotion-4 emotion-5"
>
  <div />
  <div
    className="emotion-2 emotion-3"
  >
    <button
      className="-default emotion-0 emotion-1"
      onClick={[Function]}
      type="button"
    >
      Open Menu
    </button>
  </div>
</div>
`;

exports[`<Menu /> renders menu items 1`] = `
.emotion-17 {
  position: relative;
}

.emotion-15 {
  display: inline-block;
  position: relative;
}

.emotion-13 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-13:focus,
.emotion-13:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-13:focus,
.emotion-13.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-13:disabled,
.emotion-13.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-13:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-13 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-13.-default,
.emotion-13.-primary,
.emotion-13.-danger,
.emotion-13.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-13.-default.-outline,
.emotion-13.-primary.-outline,
.emotion-13.-danger.-outline,
.emotion-13.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-13.-default.-text,
.emotion-13.-primary.-text,
.emotion-13.-danger.-text,
.emotion-13.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-13.-default:hover,
.emotion-13.-primary:hover,
.emotion-13.-danger:hover,
.emotion-13.-white:hover,
.emotion-13.-default.active,
.emotion-13.-primary.active,
.emotion-13.-danger.active,
.emotion-13.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-13.-default:hover.-outline,
.emotion-13.-primary:hover.-outline,
.emotion-13.-danger:hover.-outline,
.emotion-13.-white:hover.-outline,
.emotion-13.-default.active.-outline,
.emotion-13.-primary.active.-outline,
.emotion-13.-danger.active.-outline,
.emotion-13.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-13.-default:hover.-text,
.emotion-13.-primary:hover.-text,
.emotion-13.-danger:hover.-text,
.emotion-13.-white:hover.-text,
.emotion-13.-default.active.-text,
.emotion-13.-primary.active.-text,
.emotion-13.-danger.active.-text,
.emotion-13.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-13.-default:disabled,
.emotion-13.-primary:disabled,
.emotion-13.-danger:disabled,
.emotion-13.-white:disabled,
.emotion-13.-default.disabled,
.emotion-13.-primary.disabled,
.emotion-13.-danger.disabled,
.emotion-13.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-13.-default:disabled.-outline,
.emotion-13.-primary:disabled.-outline,
.emotion-13.-danger:disabled.-outline,
.emotion-13.-white:disabled.-outline,
.emotion-13.-default.disabled.-outline,
.emotion-13.-primary.disabled.-outline,
.emotion-13.-danger.disabled.-outline,
.emotion-13.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-13.-default:disabled.-text,
.emotion-13.-primary:disabled.-text,
.emotion-13.-danger:disabled.-text,
.emotion-13.-white:disabled.-text,
.emotion-13.-default.disabled.-text,
.emotion-13.-primary.disabled.-text,
.emotion-13.-danger.disabled.-text,
.emotion-13.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-13.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-13.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-13.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-13.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-13.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-13.-text:hover,
.emotion-13.-text.active {
  border-color: transparent;
}

.emotion-13.-text.disabled,
.emotion-13.-text:disabled {
  border: 1px solid transparent;
}

.emotion-13.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-13.disabled,
.emotion-13:disabled {
  cursor: not-allowed;
}

.emotion-12 {
  background-color: #FFF;
  border-radius: 0.25rem;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  position: absolute;
  z-index: 99999999;
  margin-top: 1rem;
  min-width: 13rem;
  padding: 0.4rem 0;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-size: 1rem;
}

.emotion-12 ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.emotion-12 li {
  margin: 0;
  padding: 0;
}

.emotion-0 {
  position: absolute;
  width: 0;
  height: 0;
  border-left: 0.5rem solid transparent;
  border-right: 0.5rem solid transparent;
  border-bottom: 0.5rem solid #FFF;
  top: -0.5rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-2 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
  border-radius: 0;
  text-align: left;
  text-transform: none;
  min-width: 150px;
  width: 100%;
}

.emotion-2:focus,
.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus,
.emotion-2.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-2:disabled,
.emotion-2.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-2:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-2 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-2.-default,
.emotion-2.-primary,
.emotion-2.-danger,
.emotion-2.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-2.-default.-outline,
.emotion-2.-primary.-outline,
.emotion-2.-danger.-outline,
.emotion-2.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-2.-default.-text,
.emotion-2.-primary.-text,
.emotion-2.-danger.-text,
.emotion-2.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-2.-default:hover,
.emotion-2.-primary:hover,
.emotion-2.-danger:hover,
.emotion-2.-white:hover,
.emotion-2.-default.active,
.emotion-2.-primary.active,
.emotion-2.-danger.active,
.emotion-2.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-2.-default:hover.-outline,
.emotion-2.-primary:hover.-outline,
.emotion-2.-danger:hover.-outline,
.emotion-2.-white:hover.-outline,
.emotion-2.-default.active.-outline,
.emotion-2.-primary.active.-outline,
.emotion-2.-danger.active.-outline,
.emotion-2.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-2.-default:hover.-text,
.emotion-2.-primary:hover.-text,
.emotion-2.-danger:hover.-text,
.emotion-2.-white:hover.-text,
.emotion-2.-default.active.-text,
.emotion-2.-primary.active.-text,
.emotion-2.-danger.active.-text,
.emotion-2.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-2.-default:disabled,
.emotion-2.-primary:disabled,
.emotion-2.-danger:disabled,
.emotion-2.-white:disabled,
.emotion-2.-default.disabled,
.emotion-2.-primary.disabled,
.emotion-2.-danger.disabled,
.emotion-2.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-2.-default:disabled.-outline,
.emotion-2.-primary:disabled.-outline,
.emotion-2.-danger:disabled.-outline,
.emotion-2.-white:disabled.-outline,
.emotion-2.-default.disabled.-outline,
.emotion-2.-primary.disabled.-outline,
.emotion-2.-danger.disabled.-outline,
.emotion-2.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-2.-default:disabled.-text,
.emotion-2.-primary:disabled.-text,
.emotion-2.-danger:disabled.-text,
.emotion-2.-white:disabled.-text,
.emotion-2.-default.disabled.-text,
.emotion-2.-primary.disabled.-text,
.emotion-2.-danger.disabled.-text,
.emotion-2.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-2.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-2.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-2.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-2.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-2.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-2.-text:hover,
.emotion-2.-text.active {
  border-color: transparent;
}

.emotion-2.-text.disabled,
.emotion-2.-text:disabled {
  border: 1px solid transparent;
}

.emotion-2.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-2.disabled,
.emotion-2:disabled {
  cursor: not-allowed;
}

.emotion-2:hover {
  background-color: #F1F1F1;
}

.emotion-8 {
  width: 100%;
  height: 0;
  border-top: 1px solid #C2C2C2;
}

<div
  className="emotion-17 emotion-18"
>
  <div>
    <div
      className="emotion-12"
      style={
        Object {
          "left": "0px",
          "top": "0px",
        }
      }
    >
      <div
        className="emotion-0 emotion-1"
        style={
          Object {
            "left": "0px",
          }
        }
      />
      <ul>
        <li>
          <button
            className="-default -text emotion-2 emotion-3"
            type="button"
          >
            Item 1
          </button>
        </li>
        <li>
          <button
            className="-default -text emotion-2 emotion-3"
            type="button"
          >
            Item 2
          </button>
        </li>
        <li>
          <button
            className="-default -text emotion-2 emotion-3"
            type="button"
          >
            Item 3
          </button>
        </li>
        <li>
          <hr
            className="emotion-8 emotion-9"
          />
        </li>
        <li>
          <button
            className="-default -text emotion-2 emotion-3"
            type="button"
          >
            Long Item 4
          </button>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="emotion-15 emotion-16"
  >
    <button
      className="-default emotion-13 emotion-3"
      onClick={[Function]}
      type="button"
    >
      Open Menu
    </button>
  </div>
</div>
`;
